cirq/contrib/quirk/cells/cell.py

```diff
@@ -11,34 +11,63 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import abc
+from typing import (Callable, Optional, List, NamedTuple, Any, Iterable,
+                    Sequence, TYPE_CHECKING, Union, Dict, Tuple)
 
-from typing import (
-    Callable,
-    Optional,
-    List,
-    NamedTuple,
-    Any,
-    Iterable,
-    Sequence,
-    TYPE_CHECKING,
-    Union,
-    Dict,
-)
-
-from cirq import ops, value
+from cirq import ops, value, devices
 
 if TYPE_CHECKING:
     import cirq
 
 
-class Cell:
+class Cell(metaclass=abc.ABCMeta):
     """A gate, operation, display, operation modifier, etc from Quirk.
 
     Represents something that can go into a column in Quirk, and supports the
     operations ultimately necessary to transform a grid of these cells into a
     `cirq.Circuit`.
     """
 
+    @classmethod
+    def _replace_qubit(cls, old_qubit: 'cirq.Qid',
+                       qubits: List['cirq.Qid']) -> 'cirq.Qid':
+        if not isinstance(old_qubit, devices.LineQubit):
```

Comment:
This is intentional. The other classes are relying on this method to enforce the type constraint. It is part of the spec of the method that it takes Qid and raises if it's not LineQubit. Otherwise we'd need many casts or rather many redundant checks elsewhere.